src/main/java/uk/gov/hmcts/reform/cwrdapi/controllers/feign/UserProfileFeignClient.java

```diff
@@ -30,11 +31,22 @@
     Response modifyUserRoles(@RequestBody UserProfileUpdatedData modifyRoles, @PathVariable("userId") String userId,
                              @RequestParam(value = "origin") String origin);
 
-
     @GetMapping(value = "/v1/userprofile/{id}/roles")
     @RequestLine("GET /v1/userprofile/{id}/roles")
     @Headers({"Authorization: {authorization}", "ServiceAuthorization: {serviceAuthorization}",
         "Content-Type: application/json"})
     Response getUserProfileWithRolesById(@PathVariable String id);
 
+    @DeleteMapping(value = "/v1/userprofile/users?userId={userId}")
+    @RequestLine("DELETE /v1/userprofile/users?userId={userId}")
+    @Headers({"Authorization: {authorization}", "ServiceAuthorization: {serviceAuthorization}",
+        "Content-Type: application/json"})
+    Response deleteUserProfileById(@PathVariable String userId);
+
+    @DeleteMapping(value = "/v1/userprofile/users?emailPattern={emailPattern}")
+    @RequestLine("DELETE /v1/userprofile/users?emailPattern={emailPattern}")
+    @Headers({"Authorization: {authorization}", "ServiceAuthorization: {serviceAuthorization}",
+        "Content-Type: application/json"})
```

Comment:
1 ) how Rest call will be differentiated here ? Delete  /v1/userprofile/users will not clash ? i think you can keep only one call here
2) These is no @PathVariable here , i think only has request param

Reply:
Addressed here: https://github.com/hmcts/rd-caseworker-ref-api/pull/287/commits/39bd1ddb47947204d4427c142785dfb2036ade12